### PR TITLE
[SYCL] Enable LIT testing for Level Zero

### PR DESCRIPTION
**Pull request by againull**
* Support discovery of Level Zero devices in get_device_count_by_type tool
* Support Level Zero backend in the lit scripts
* Add check-level0 make target
* Fix obvious mistake in the plugin in addMapping function
* Disable lit tests failing for Level Zero

### CODE REVIEW

File: sycl/test/lit.cfg.py [hunk 1]
```diff
@@ -105,10 +106,16 @@ def getDeviceCount(device_type):
         if re.match(r".*cuda", result[1]):
             is_cuda = True;
 
+    # if we have found gpu and there is additional information, let's check
+    # whether this is CUDA device or not
```
Comment (bjoernknafla) [inline sycl/test/lit.cfg.py:110]:
```suggestion
    # whether this is a Level0 device or not
```

File: sycl/test/lit.cfg.py [hunk 2]
```diff
@@ -105,10 +106,16 @@ def getDeviceCount(device_type):
...
+    if device_type == "gpu" and value > 0 and len(result[1]):
```
Comment (bjoernknafla) [inline sycl/test/lit.cfg.py:111]: Would it make sense to also move the CUDA check above into this condition?

**Comment by bjoernknafla** [comment]
+1

File: sycl/test/lit.cfg.py (hunk 2 above)
Comment (againull) [reply sycl/test/lit.cfg.py]: You are right, fixed.

**Comment by bjoernknafla** [comment]
The CUDA build is probably fixed by commit 4a8c4322e976529a0ffe86e628a32dfc6969f048 ?

File: sycl/test/lit.cfg.py [hunk 3]
```diff
@@ -100,15 +101,17 @@ def getDeviceCount(device_type):
...
         if re.match(r".*cuda", result[1]):
             is_cuda = True;
+        if re.match(r".*level zero", result[1]):
+            is_level0 = True;
 
     if err:
         lit_config.warning("getDeviceCount {TYPE} {BACKEND} stderr:{ERR}".format(
             TYPE=device_type, BACKEND=backend, ERR=err))
-    return [value,is_cuda]
+    return [value,is_cuda,is_level0]
```
Comment (bader) [inline sycl/test/lit.cfg.py:114]: Please, fix return value type at line 93.
https://github.com/intel/llvm/runs/760440987

**Pull request merged**